core/src/main/java/io/grpc/internal/ServerTransportListener.java

```diff
@@ -40,6 +40,13 @@
 public interface ServerTransportListener {
 
   /**
+   * Called when the method name for a new stream has been determined.
+   *
+   * @return a context object for recording stats and tracing for the new stream.
+   */
+  StatsTraceContext methodDetermined(String methodName, Metadata headers);
```

Comment:
I find the name a bit awkward as well, I guess because I don't really know when it happens. Maybe include a comment that it occurs before `streamCreated()`?